integration_tests/runJest.js

```diff
@@ -26,7 +43,16 @@ function runJest(dir, args) {
 runJest.json = function(dir, args) {
   args = Array.prototype.concat(args || [], '--json');
   const result = runJest(dir, args);
-  result.json = JSON.parse((result.stdout || '').toString());
+  try {
+    result.json = JSON.parse((result.stdout || '').toString());
+  } catch (e) {
+    throw new Error(`
+      Can't parse JSON.
```

Comment:
these messages do `      ` as indentation. Is that expected? (Also above for the infinite loop error).